spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -73,14 +73,22 @@ statement
     | ALTER TABLE multipartIdentifier WRITE writeSpec                                       #setWriteDistributionAndOrdering
     | ALTER TABLE multipartIdentifier SET IDENTIFIER_KW FIELDS fieldList                    #setIdentifierFields
     | ALTER TABLE multipartIdentifier DROP IDENTIFIER_KW FIELDS fieldList                   #dropIdentifierFields
-    | ALTER TABLE multipartIdentifier createReplaceBranchClause   #createOrReplaceBranch
+    | ALTER TABLE multipartIdentifier createReplaceBranchClause                             #createOrReplaceBranch
+    | ALTER TABLE multipartIdentifier createTagClause                                       #createTag
+    ;
+
+createTagClause
+    : CREATE TAG (IF NOT EXISTS)? identifier tagOptions
     ;
 
 createReplaceBranchClause
     : (CREATE OR)? REPLACE BRANCH identifier branchOptions
     | CREATE BRANCH (IF NOT EXISTS)? identifier branchOptions
     ;
 
+tagOptions
+    : (AS OF VERSION snapshotId)? (refRetain)?;
```

Comment:
hmm I just noticed this `;` in the end, also in `branchOptions`. @amogh-jahagirdar is this required? It doesn't feel like required to me.

Reply:
this is required. We can insert a new line, consistent with the rest.